src/java/org/apache/cassandra/schema/TableMetadata.java

```diff
@@ -602,11 +606,16 @@ public boolean dependsOn(Function function)
 
     public void validate()
     {
-        if (!isNameValid(keyspace))
-            except("Keyspace name must not be empty, more than %s characters long, or contain non-alphanumeric-underscore characters (got \"%s\")", SchemaConstants.NAME_LENGTH, keyspace);
+        if (!isValidKeyspaceName(keyspace))
+            except("Keyspace name must not be empty, more than %s characters long, or contain non-alphanumeric-underscore characters (got \"%s\")", NAME_LENGTH, keyspace);
 
-        if (!isNameValid(name))
-            except("Table name must not be empty, more than %s characters long, or contain non-alphanumeric-underscore characters (got \"%s\")", SchemaConstants.NAME_LENGTH, name);
+        if (!isValidName(name))
```

Comment:
@k-rus why not to have "isValidTableName" method when we have already "isValidKeyspaceName" method? If we put "keyspace" in one, we should put "table" in the other.

Reply:
If "isValidName" is used at other places too, we could have "isValidTableName()" method in TableMetadata which would just call "isValidName" in SchemaConstants. It seems like an overkill but I think that having the naming consistent is worth of it.